src/Dto/SpecialSponsorshipRequest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Dto;
+
+use App\Entity\SpecialSponsorship;
+use Symfony\Component\Validator\Constraints as Assert;
+
+final class SpecialSponsorshipRequest
+{
+    /**
+     * @Assert\NotBlank()
+     * @Assert\Type(type="string")
+     */
+    public $label;
+
+    /**
+     * @Assert\NotBlank()
+     * @Assert\Type(type="float")
+     */
+    public $price;
+
+    /**
+     * @Assert\NotBlank()
+     */
+    public $description;
+
+    public function __construct(string $label = null, float $price = null, string $description = null)
```

Comment:
`$description` has an `@Assert\NotBlank()` above, which will it be?

Reply:
I don't understand what you mean here, Should the description not be empty ?

Reply:
I think it's because you define a default value as null in the function parameter list, and as the same time you define a NotBlank validator. So it's not logic.

Reply:
Ok but to create a SpecialBenefit for example, we need to do new SpecialBenefitRequest so i need that all the datas are empty

Reply:
Please have a look at [Webmozart's take on SF forms with VO/DTO](https://webmozart.io/blog/2015/09/09/value-objects-in-symfony-forms/), it sums up exaclty what nicer options you may have to keep your object in a consistent state